client/my-sites/site-indicator/site-indicator.jsx

```diff
@@ -295,4 +357,32 @@ export default React.createClass( {
 			</div>
 		);
 	}
-} );
+
+	render() {
+		const { site, siteIsJetpack } = this.props;
+
+		return (
+			<div className="site-indicator__wrapper">
+				{ siteIsJetpack && <QuerySiteConnectionStatus siteId={ site.ID } /> }
```

Comment:
Maybe consider passing through isJetpack as a prop, and updating the query component to do a no-op if it's falsey.
```
<QuerySiteConnectionStatus siteId={ site.ID } isJetpack />
```